### DISCUSSION TITLE
Stuck building app

### DISCUSSION THREAD

**Discussion by capidreamliner** [flet-dev/flet#2426] [Q&A]
### Question

Hello everyone,

I'm new to Flet and facing a bit of trouble that's been bothering me for a while now.

I'm trying to build my android app, and I followed the guide in the documentation, installed Flutter, and ran the 'flutter doctor' command without any problems. The app builds fine as long as I only have Flet in the requirements. However, my app needs either pyrebase or firebase-rest-api, and I haven't been able to get it to build with either of them included.

I've been stuck on this problem for two days now, trying different Flet versions (0.18.0, 0.19.0, 0.20.0). The higher versions get stuck in "Packaging Python App," and the one that got me the farthest was 0.18.0. However, I encountered an error during the "Building APK for Android" phase, which I've included in the code section below.

This issue isn't just on Android; I've also been unable to export my app with this library to any other platform.

I'm sorry for bothering, and I really appreciate Flet. I would love to use it for my app, but I'm facing this difficulty that holds me back. Any help or tips would be greatly appreciated.

Thank you so much in advance for any help. Your work on Flet is really admirable.

### Code sample

```python
Code (Test code):

import flet as ft
import firebase

firebaseConfig = {

    SENSITIVE DATA

}

app = firebase.initialize_app(firebaseConfig)

auth = app.auth()
db = app.database()

email = 'SENSITIVE DATA'
password = 'SENSITIVE DATA'

user = auth.sign_in_with_email_and_password(email, password)

def main(page: ft.Page):
    page.add(ft.SafeArea(ft.Text(str(user))))


ft.app(main)

REQUIREMENTS:

flet
firebase-rest-api


```

### Error message

```bash
Creating Flutter bootstrap project...OK
Customizing app icons and splash images...OK
Generating app icons...OK
Generating splash screens...OK
Packaging Python app...OK
Building .apk for Android...Resolving dependencies...
  ffi 2.1.0 (2.1.2 available)
  fl_chart 0.65.0 (0.66.0 available)
  flutter_lints 2.0.3 (3.0.1 available)
  flutter_native_splash 2.3.7 (2.3.9 available)
  js 0.6.7 (0.7.0 available)
  lints 2.1.1 (3.0.0 available)
  matcher 0.12.16 (0.12.16+1 available)
  material_color_utilities 0.5.0 (0.8.0 available)
  meta 1.10.0 (1.11.0 available)
  package_info_plus 4.2.0 (5.0.1 available)
  path 1.8.3 (1.9.0 available)
  petitparser 5.4.0 (6.0.2 available)
  test_api 0.6.1 (0.7.0 available)
  toml 0.14.0 (0.15.0 available)
  web 0.3.0 (0.4.2 available)
  web_socket_channel 2.4.0 (2.4.3 available)
  xml 6.3.0 (6.5.0 available)
Got dependencies!
17 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.

Running Gradle task 'assembleRelease'...
Error detected in pubspec.yaml:
Running Gradle task 'assembleRelease'...                           87.1s

C:\Users\nsqse\AppData\Local\Android\sdk\ndk\23.1.7779620\toolchains\llvm\prebuilt\windows-x86_64\bin\llvm-strip.exe: error:
'C:\Users\nsqse\AppData\Local\Temp\flet_flutter_build_xUhAEwYJrW\build\serious_python_android\intermediates\merged_native_libs\release\out\lib\armeabi-v7a\libpythonbundle.so': The 
file was not recognized as a valid object file

C:\Users\nsqse\AppData\Local\Android\sdk\ndk\23.1.7779620\toolchains\llvm\prebuilt\windows-x86_64\bin\llvm-strip.exe: error:
'C:\Users\nsqse\AppData\Local\Temp\flet_flutter_build_xUhAEwYJrW\build\serious_python_android\intermediates\merged_native_libs\release\out\lib\arm64-v8a\libpythonbundle.so': The   
file was not recognized as a valid object file

C:\Users\nsqse\AppData\Local\Android\sdk\ndk\23.1.7779620\toolchains\llvm\prebuilt\windows-x86_64\bin\llvm-strip.exe: error:
'C:\Users\nsqse\AppData\Local\Temp\flet_flutter_build_xUhAEwYJrW\build\serious_python_android\intermediates\merged_native_libs\release\out\lib\x86_64\libpythonbundle.so': The file 
was not recognized as a valid object file

No file or variants found for asset: app/app.zip.hash.

Target aot_android_asset_bundle failed: Exception: Failed to bundle asset files.


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileFlutterBuildRelease'.
> Process 'command 'C:\Users\nsqse\flutter\flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 26s
Gradle task assembleRelease failed with exit code 1

Error building Flet app - see the log of failed command above.
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment]
Upgrade to Flet 0.19.0 to eliminate `No file or variants found for asset: app/app.zip.hash` error.

**Reply by capidreamliner** [reply]
Hello Feodor!

Firstly, thank you very much for your help and for the excellent work you are doing with flet. Really admirable. 

As mentioned above, when I use flet 0.19.0 the process stays at 

> Packaging Python app...

![image](https://github.com/flet-dev/flet/assets/156849492/843f4060-fe18-4604-8834-eb4e6239d072)


I left my computer running last night thinking it might take time, no luck. No errors, it just stays in that stage.

Thank you very much, I look forward to your reply

**Comment by FeodorFitsner** [comment]
Please run with `flet build apk -vv` and provide a full log.

**Comment by Humarr** [comment]
Hey Feodor, I am having the same issue as him. It gets stuck at "packaging python app...". 

Going with your instruction to run with `flet build apk -vv`, I got the following logs: 

PS G:\My_projects\FletPython\finatrack> flet build apk -vv
Flutter bootstrap directory: 
C:\Users\NICEGU~1\AppData\Local\Temp\flet_flutter_build_PqYgwrZaQu
Creating Flutter bootstrap project...OK
Customizing app icons and splash images...Copying G:\My_projects\FletPython\finatrack\assets\icon_android.png to
C:\Users\NICEGU~1\AppData\Local\Temp\flet_flutter_build_PqYgwrZaQu\images
Copying G:\My_projects\FletPython\finatrack\assets\splash_android.png to
C:\Users\NICEGU~1\AppData\Local\Temp\flet_flutter_build_PqYgwrZaQu\images
OK
Generating app icons...Resolving dependencies in C:\Users\NICEGU~1\AppData\Local\Temp\flet_flutter_build_PqYgwrZaQu... (7.6s)
Got dependencies in C:\Users\NICEGU~1\AppData\Local\Temp\flet_flutter_build_PqYgwrZaQu.
Building package executable... (9.8s)
Built flutter_launcher_icons:flutter_launcher_icons.
  ════════════════════════════════════════════
     FLUTTER LAUNCHER ICONS (v0.13.1)
  ════════════════════════════════════════════

• Creating default icons Android
• Overwriting the default Android launcher icon with a new icon
• Overwriting default iOS launcher icon with new icon
Creating Icons for Web...
Creating Icons for Windows...
Creating Icons for MacOS...

✓ Successfully generated launcher icons
OK
Generating splash screens...Building package executable... (10.7s)
Built flutter_native_splash:create.
[Android] Creating default splash images
[Android] Creating dark mode splash images
[Android] Creating default android12splash images
[Android] Creating dark mode android12splash images
[Android] Updating launch background(s) with splash image path...
[Android]  - android/app/src/main/res/drawable/launch_background.xml
[Android]  - android/app/src/main/res/drawable-night/launch_background.xml
[Android]  - android/app/src/main/res/drawable-v21/launch_background.xml
[Android]  - android/app/src/main/res/drawable-night-v21/launch_background.xml
[Android] Updating styles...
[Android]  - android/app/src/main/res/values-v31/styles.xml
[Android] No android/app/src/main/res/values-v31/styles.xml found in your Android project
[Android] Creating android/app/src/main/res/values-v31/styles.xml and adding it to your Android project
[Android]  - android/app/src/main/res/values-night-v31/styles.xml
[Android] No android/app/src/main/res/values-night-v31/styles.xml found in your Android project
[Android] Creating android/app/src/main/res/values-night-v31/styles.xml and adding it 
to your Android project
[Android]  - android/app/src/main/res/values/styles.xml
[Android]  - android/app/src/main/res/values-night/styles.xml
[iOS] Creating  images
[iOS] Creating dark mode  images
[iOS] Updating ios/Runner/Info.plist for status bar hidden/visible
[Web] Creating images
[Web] Creating images
[Web] Creating background images
[Web] Creating CSS
[Web] Updating index.html
╔════════════════════════════════════════════════════════════════════════════╗
║                       NEED A GREAT FLUTTER DEVELOPER?                      ║        
╠════════════════════════════════════════════════════════════════════════════╣        
║                                                                            ║        
║   I am available!  Find me at https://www.linkedin.com/in/hansonjon/       ║        
║                                                                            ║        
╚════════════════════════════════════════════════════════════════════════════╝        

✅ Native splash complete.
Now go finish building something awesome! 💪 You rock! 🤘🤩
Like the package? Please give it a 👍 here: https://pub.dev/packages/flutter_native_splash

OK
Packaging Python app...Building package executable... (8.8s)
Built serious_python:main.
Running package command
Creating asset directory: C:\Users\NICEGU~1\AppData\Local\Temp\flet_flutter_build_PqYgwrZaQu\app
Copying Python app from G:\My_projects\FletPython\finatrack to C:\Users\NICEGU~1\AppData\Local\Temp\serious_python_temp21cb6027
Configured mobile platform with sitecustomize.py at C:\Users\NICEGU~1\AppData\Local\Temp\serious_python_sitecustomize34629e80\sitecustomize.py
Installing dependencies [flet-embed, firebase-admin, requests, beautifulsoup4, kvdroid, pyjnius, plyer] with pip command to C:\Users\NICEGU~1\AppData\Local\Temp\serious_python_temp21cb6027\__pypackages__
Extracting Python distributive from C:\Users\NICEGU~1\AppData\Local\Temp\cpython-3.11.6+20231002-x86_64-pc-windows-msvc-shared-install_only.tar.gz to C:\Users\NICEGU~1\AppData\Local\Temp\hostpython3.11_7fff62f8
VERBOSE: C:\Users\NICEGU~1\AppData\Local\Temp\hostpython3.11_7fff62f8\python\python.exe -m pip install --isolated --upgrade --target C:\Users\NICEGU~1\AppData\Local\Temp\serious_python_temp21cb6027\__pypackages__ flet-embed firebase-admin requests beautifulsoup4 kvdroid pyjnius plyer
VERBOSE: Collecting flet-embed
VERBOSE: Obtaining dependency information for flet-embed from https://files.pythonhosted.org/packages/ad/d1/a6b49f4fae4611d8b083da1f95afd78e38ffb3898290ffc8dcec73324bf6/flet_embed-0.19.0-py3-none-any.whl.metadata
VERBOSE: Using cached flet_embed-0.19.0-py3-none-any.whl.metadata (1.1 kB)
VERBOSE: Collecting firebase-admin
VERBOSE: Obtaining dependency information for firebase-admin from https://files.pythonhosted.org/packages/04/5b/b70cbdbddd36d52e16218931a591783871fb5b307b9449ad8efff14eb0f4/firebase_admin-6.4.0-py3-none-any.whl.metadata
VERBOSE: Using cached firebase_admin-6.4.0-py3-none-any.whl.metadata (1.5 kB)
VERBOSE: Collecting requests
VERBOSE: Obtaining dependency information for requests from https://files.pythonhosted.org/packages/70/8e/0e2d847013cb52cd35b38c009bb167a1a26b2ce6cd6965bf26b47bc0bf44/requests-2.31.0-py3-none-any.whl.metadata
VERBOSE: Using cached requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)
VERBOSE: Collecting beautifulsoup4
VERBOSE: Obtaining dependency information for beautifulsoup4 from https://files.pythonhosted.org/packages/b1/fe/e8c672695b37eecc5cbf43e1d0638d88d66ba3a44c4d321c796f4e59167f/beautifulsoup4-4.12.3-py3-none-any.whl.metadata
VERBOSE: Using cached beautifulsoup4-4.12.3-py3-none-any.whl.metadata (3.8 kB)
VERBOSE: Collecting kvdroid
VERBOSE: Using cached kvdroid-0.3.0-py3-none-any.whl
VERBOSE: Collecting pyjnius
VERBOSE: Using cached pyjnius-1.6.1.tar.gz (63 kB)
VERBOSE: Installing build dependencies: started
VERBOSE: Installing build dependencies: finished with status 'done'
VERBOSE: Getting requirements to build wheel: started
VERBOSE: Getting requirements to build wheel: finished with status 'error'

**Comment by Humarr** [comment]
It shows that, the process got stuck due to some errors that may occur during the build process. However, Why does it fail to build pyjnius but builds all my other dependencies: 
`flet`
`firebase-admin`
`requests`
`beautifulsoup4`
`kvdroid`
`pyjnius` (build fails here)
`plyer`

**Reply by wandering3ngineer** [reply]
I had a similar problem. I have a docker deployed desktop ubuntu dev environment. When I was using the terminal built into VSCodium it regularly hung on "flet build" execution. However, opening a dedicated terminal and closing down unnecessary apps led to a successful build. Not sure of underlying reason alas.

**Comment by Muddassir-Farooq-official** [comment]
Is flet support plyer??

**Reply by ndonkoHenri** [reply]
https://flet.dev/blog/tap-into-native-android-and-ios-apis-with-Pyjnius-and-pyobjus/#plyer-challenge

**Comment by Muddassir-Farooq-official** [comment]
> Please run with `flet build apk -vv` and provide a full log.

Still waiting for any update or hope https://github.com/flet-dev/flet/issues/5166
